libp2p/muxers/mplex/lpchannel.nim

```diff
@@ -93,12 +93,45 @@ proc resetMessage(s: LPChannel) {.async.} =
   await s.conn.writeMsg(s.id, s.resetCode)
 
 proc resetByRemote*(s: LPChannel) {.async.} =
-  await allFutures(s.close(), s.closedByRemote())
+  let
+    f1 = awaitne s.close()
+    f2 = awaitne s.closedByRemote()
+  # NOTICE WE DO NOT RETHROW
+  # Those failures are not critical but still wait and warn about them!
+  if f1.failed:
+    let exc = f1.readError()
+    # EOF are likely to happen here
+    if exc.name != "LPStreamEOFError":
```

Comment:
I'm kind of surprised `exc of (ref LPStreamEOFError)` doesn't work here - was this where you tried it?

Reply:
Yep but I made it compile at some point tho, I will review it.

Reply:
Hmm not sure why you see this, but its already fixed btw indeed, you linked an outdated one

Reply:
btw don't need `(ref LPStreamEOFError)` , just `LPStreamEOFError`

Reply:
yeah - that's a bit odd in my book, but sure, whatever works ;)

Reply:
hmm why? it's a `ref object` kind of type

Reply:
Yes, I'm too surprised that this didn't work. I've verified that `traceAsyncErrors` for example works and it's based on the `of` operator. Perhaps you need to dereference the exception like I did here:

https://github.com/status-im/nim-eth/blob/b3b5854003ebb10b99efa451d4391ef050f5951e/eth/async_utils.nim#L14

Reply:
@zah it worked, I just commented on an old version of the code.. that said, if `exc` is a `ref LPStreamEOFError` and matches `of LPStreamEOFError` , it means the `of` operator is more lenient than I would have expected it to be..